jetty-proxy/src/main/java/org/eclipse/jetty/proxy/ConnectHandler.java

```diff
@@ -589,6 +592,38 @@ protected void write(EndPoint endPoint, ByteBuffer buffer, Callback callback)
         {
             ConnectHandler.this.write(endPoint, buffer, callback, getContext());
         }
+
+        private class OnCompleteListener implements AsyncListener
+        {
+            private final Runnable onComplete;
+
+            private OnCompleteListener(Runnable onComplete)
```

Comment:
This feels needlessly generic?   Why not just make the `UpstreamConnection` implement `AsyncListener`, then it can just call `fillInterest` in `onComplete`?